django-verdant/rca/management/commands/archive_events.py

```diff
@@ -0,0 +1,161 @@
+        ),
+        make_option(
+            '--csv',
+            action='store_true',
+            dest='output_csv',
+            default=False,
+            help="List archived event items. Negates verbose."
+        ),
+        make_option(
+            '--permanent',
+            action='store_true',
+            dest='permanent',
+            default=True,
+            help="Create permanent redirects (301)."
+        ),
+    )
+
+    def create_redirect(self, e, target_page, permanent=True):
+        new_redirect = Redirect()
+        new_redirect.site = e.get_site()
+        # Path includes 'home' and a trailing slash
+        # both must be removed.
+        bare_path = '/' + '/'.join(e.url_path.split('/')[2:][:-1])
+        new_redirect.old_path = bare_path
+        new_redirect.is_permanent = permanent
+        new_redirect.redirect_page = target_page
+        new_redirect.save()
+        self.event_redirects[e.pk] = new_redirect.id
+
+    def get_csv_rows(self, events):
+        rows = []
+        headers = [
+            'Event id',
+            'Event title',
+            'Event URL',
+            'Dates from',
+            'Dates to',
+            'Redirect',
+        ]
+        rows.append(headers)
+        for e in events:
+            row = []
+            row.append(e.pk)
+            row.append(e.title.encode('utf8'))
+            row.append(e.url)
+            dates = e.dates_times.order_by('date_from').values_list('date_from', 'date_to')
+            date_from = []
+            for d in dates:
+                try:
+                    date_from.append(d[0].strftime("%d/%m/%Y"))
+                except Exception:
+                    date_from = ''
+
+            date_to = []
+            for d in dates:
+                try:
+                    date_to.append(d[1].strftime("%d/%m/%Y"))
+                except Exception:
+                    date_to = ''
+
+            row.append(', '.join(date_from))
+            row.append(', '.join(date_to))
+            if self.event_redirects:
+                row.append('{}/admin/redirects/{}/'.format(
+                    e.get_site().root_url,
+                    self.event_redirects[e.pk])
+                )
+            else:
+                row.append('')
+            rows.append(row)
+
+        return rows
+
+    def handle(self, **options):
+        verbose = options['verbose']
+        examples = options['examples']
+        output_csv = options['output_csv']
+        permanent = options['permanent']
+        dry_run = options['dry_run']
+
+        event_index = EventIndex.objects.first()
+
+        if csv:
+            verbose = False
+            examples = False
+
+        events = EventItem.objects.live()
+        events_count = events.count()
+        two_years_ago = date.today() - relativedelta(years=2)
+
+        events = events.exclude(dates_times__date_to__gt=two_years_ago)
+        events = events.exclude(dates_times__date_from__gt=two_years_ago)
```

Comment:
You should be able to replace these 2 lines with:
```
from django.db.models import Q
events.exclude(Q(dates_times__date_to__gt=two_years_ago) | Q(dates_times__date_from__gt=two_years_ago))
```

https://docs.djangoproject.com/en/2.2/topics/db/queries/#complex-lookups-with-q-objects

Reply:
Does that offer an advantage? I would have thought the two would be represented the same when evaluated.

Reply:
I tested it and the counts were the same, so the advantages are fewer lines of code and, more importantly, better readability as you only need to comprehend one `exclude()` call (there may be a speed advantage, don't know tbh).